File: smash-ranks-android/app/src/main/java/com/garpr/android/preferences/persistent/PersistentBooleanPreference.java
package com.garpr.android.preferences.persistent;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.garpr.android.preferences.KeyValueStore;

public class PersistentBooleanPreference extends BasePersistentPreference<Boolean> {

    public PersistentBooleanPreference(@NonNull final String key,
            @Nullable final Boolean defaultValue, @NonNull final KeyValueStore keyValueStore) {
        super(key, defaultValue, keyValueStore);
    }

    @Nullable
    @Override
    public Boolean get() {
        if (hasValueInStore()) {
            // at this point, returning the fallback value is impossible
            return getKeyValueStore().getBoolean(getKey(), false);
        } else {
            return getDefaultValue();
        }
    }

    @Override
    protected void performSet(@NonNull final Boolean newValue, final boolean notifyListeners) {
        getKeyValueStore().setBoolean(getKey(), newValue);

        if (notifyListeners) {
            notifyListeners();
        }
    }

}

File: smash-ranks-android/app/src/main/java/com/garpr/android/preferences/persistent/PersistentBooleanPreference.kt
package com.garpr.android.preferences.persistent

import com.garpr.android.preferences.KeyValueStore

class PersistentBooleanPreference(
        key: String,
        defaultValue: Boolean?,
        keyValueStore: KeyValueStore
) : BasePersistentPreference<Boolean>(
        key,
        defaultValue,
        keyValueStore
) {

    override fun get(): Boolean? {
        return if (hasValueInStore) {
            // at this point, returning the fallback value is impossible
            keyValueStore.getBoolean(key, false)
        } else {
            defaultValue
        }
    }

    override fun performSet(newValue: Boolean) {
        keyValueStore.setBoolean(key, newValue)
        notifyListeners()
    }

}
